lib/tests/test_rewrite.rs

```diff
@@ -1612,3 +1612,76 @@ fn test_empty_commit_option(empty_behavior: EmptyBehaviour) {
         }
     );
 }
+
+#[test]
+fn test_rebase_commit_via_rebase_api() {
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    // Rebase B onto B2
+    //
+    // D (empty)  E (WC, empty)
+    // |         /
+    // C---------
+    // |
+    // B B2
+    // |/
+    // A
+    let mut tx = repo.start_transaction(&settings);
+    let commit_a = write_random_commit(tx.mut_repo(), &settings);
+    let commit_b = create_random_commit(tx.mut_repo(), &settings)
+        .set_parents(vec![commit_a.id().clone()])
+        .write()
+        .unwrap();
+    let commit_c = create_random_commit(tx.mut_repo(), &settings)
+        .set_parents(vec![commit_b.id().clone()])
+        .write()
+        .unwrap();
+    let commit_d = create_random_commit(tx.mut_repo(), &settings)
+        .set_parents(vec![commit_c.id().clone()])
+        .set_tree_id(commit_c.tree_id().clone())
+        .write()
+        .unwrap();
+    let commit_e = create_random_commit(tx.mut_repo(), &settings)
+        .set_parents(vec![commit_c.id().clone()])
+        .set_tree_id(commit_c.tree_id().clone())
+        .write()
+        .unwrap();
+    let commit_b2 = create_random_commit(tx.mut_repo(), &settings)
```

Comment:
You could add a comment: `// Commit B2 has the same tree as B`. The comment could go here or in the above comment.